core/rawdb/database_quorum.go

```diff
@@ -49,10 +53,23 @@ func GetPrivateStateRoot(db ethdb.Database, blockRoot common.Hash) common.Hash {
 	return common.BytesToHash(root)
 }
 
+func GetPrivacyMetadataStateRootForPrivateStateRoot(db ethdb.KeyValueReader, privateStateRoot common.Hash) common.Hash {
+	root, _ := db.Get(append(privateRootToPrivacyMetadataRootPrefix, privateStateRoot[:]...))
+	return common.BytesToHash(root)
+}
+
 func WritePrivateStateRoot(db ethdb.Database, blockRoot, root common.Hash) error {
 	return db.Put(append(privateRootPrefix, blockRoot[:]...), root[:])
 }
 
+func WritePrivacyMetadataStateRootForPrivateStateRoot(db ethdb.KeyValueWriter, privateStateRoot, privacyMetadataRoot common.Hash) error {
+	err := db.Put(append(privateRootToPrivacyMetadataRootPrefix, privateStateRoot[:]...), privacyMetadataRoot[:])
+	if err != nil {
+		return fmt.Errorf("unable to persist mapping between private state root to privacy metadata root. Cause: %v", err)
```

Comment:
We should not wrap the error with the message here, instead just return whatever `db.Put()` returns. The calling function should provide helpful log if required

Reply:
fixed